### PR TITLE
adding truth checkers

### PR DESCRIPTION
**Pull request by dustinvogt**

### CODE REVIEW

File: src/smidje/parser/parser.clj [hunk 1]
```diff
@@ -1,5 +1,6 @@
 (ns smidje.parser.parser
-  (:require [smidje.parser.arrows :refer [arrow-set]]))
+  (:require [smidje.parser.arrows :refer [arrow-set]]
+            [smidje.parser.checkers :refer :all]))
```
Comment (ragnorak1000) [inline src/smidje/parser/parser.clj]: I've had problems with test reloading when using :refer :all, I would recommend either using :as or referring each of the checkers explicitly

File: src/smidje/parser/checkers.clj [hunk 2]
```diff
@@ -0,0 +1,12 @@
+(ns smidje.parser.checkers)
```
Comment (ragnorak1000) [inline src/smidje/parser/checkers.clj]: does it make sense to group provided and throws with checkers?

Comment (dustinvogt) [inline src/smidje/parser/checkers.clj]: throws is definitely a checker. I could see pulling leaving provided in parser for now.

File: src/smidje/parser/parser.clj (hunk 1 above)
Comment (dustinvogt) [inline src/smidje/parser/parser.clj]: I could refer individual elements. I don't want to stick :as because of readability oddness.

**Comment by ragnorak1000** [comment]
can you update the data-model.md file with :truth-testing and :throws-exception. I wan't to keep that as up to date as possible (you might want to merge in master first to avoid conflicts).

**Comment by djljr** [comment]
Should truthy and falsey be implemented as a special case of checkers now that those are in place?

**Pull request merged**